spec/requests/custom_actions_spec.rb

```diff
@@ -220,4 +220,471 @@ def expect_result_to_have_custom_actions_hash
+      )
+    end
+
+    it "accept custom actions" do
+      api_basic_authorize
+
+      post api_security_group_url(nil, @resource), :params => gen_request(:button1, "key1" => "value1")
+
+      expect_single_action_result(:success => true, :message => /.*/, :href => api_security_group_url(nil, @resource))
+    end
+  end
+
+  describe "Storage" do
+    before(:each) do
+      @resource = FactoryGirl.create(:storage)
+      define_custom_button1(@resource)
+    end
+
+    it "queries return custom actions defined" do
+      api_basic_authorize(action_identifier(:data_stores, :read, :resource_actions, :get))
+
+      get api_data_store_url(nil, @resource)
+
+      expect(response.parsed_body).to include(
+        "id"      => @resource.id.to_s,
+        "href"    => api_data_store_url(nil, @resource),
+        "actions" => a_collection_including(a_hash_including("name" => "button1"))
+      )
+    end
+
+    it "accept custom actions" do
+      api_basic_authorize
+
+      post api_data_store_url(nil, @resource), :params => gen_request(:button1, "key1" => "value1")
+
+      expect_single_action_result(:success => true, :message => /.*/, :href => api_data_store_url(nil, @resource))
+    end
+  end
+
+  describe "Tenant" do
+    before(:each) do
+      @resource = FactoryGirl.create(:tenant)
+      define_custom_button1(@resource)
+    end
+
+    it "queries return custom actions defined" do
+      api_basic_authorize(action_identifier(:tenants, :read, :resource_actions, :get))
+
+      get api_tenant_url(nil, @resource)
+
+      expect(response.parsed_body).to include(
+        "id"      => @resource.id.to_s,
+        "href"    => api_tenant_url(nil, @resource),
+        "actions" => a_collection_including(a_hash_including("name" => "button1"))
+      )
+    end
+
+    it "accept custom actions" do
+      api_basic_authorize
+
+      post api_tenant_url(nil, @resource), :params => gen_request(:button1, "key1" => "value1")
+
+      expect_single_action_result(:success => true, :message => /.*/, :href => api_tenant_url(nil, @resource))
+    end
+  end
+
+  describe "User" do
+    before(:each) do
+      @resource = FactoryGirl.create(:user)
+      define_custom_button1(@resource)
+    end
+
+    it "queries return custom actions defined" do
+      api_basic_authorize(action_identifier(:users, :read, :resource_actions, :get))
+
+      get api_user_url(nil, @resource)
+
+      expect(response.parsed_body).to include(
+        "id"      => @resource.id.to_s,
+        "href"    => api_user_url(nil, @resource),
+        "actions" => a_collection_including(a_hash_including("name" => "button1"))
+      )
+    end
+
+    it "accept custom actions" do
+      api_basic_authorize
+
+      post api_user_url(nil, @resource), :params => gen_request(:button1, "key1" => "value1")
+
+      expect_single_action_result(:success => true, :message => /.*/, :href => api_user_url(nil, @resource))
+    end
+  end
+
+  describe "Vms" do
+    before(:each) do
```

Comment:
another NIT, probably just need `before`, not `before(:each)` (another thing I'll have to update in my own 😛 )